File: services/auth/Security/HashIds.cs
using HashidsNet;

namespace Bacera.Gateway.Auth.Security;

/// <summary>
/// Mirrors hashids.rs — encodes Party/User IDs using the same salts and alphabets as mono.
/// </summary>
public static class HashIds
{
    private const string PartySalt = "BCRPartyId";
    private const string PartyAlphabet = "ABCDEFGHJKMNPQRSTUVWXYZ0123456789";
    private const string UserSalt = "BCRUserId";
    private const string UserAlphabet = "ABCDEFGHJKMNPQRSTUVWXYZ0123456789";
    private const int MinLength = 8;

    private static readonly Hashids PartyHashids = new(PartySalt, MinLength, PartyAlphabet);
    private static readonly Hashids UserHashids = new(UserSalt, MinLength, UserAlphabet);

    public static string EncodePartyId(long id) => PartyHashids.EncodeLong(id);
    public static long DecodePartyId(string hash) => PartyHashids.DecodeLong(hash).FirstOrDefault();

    public static string EncodeUserId(long id) => UserHashids.EncodeLong(id);
    public static long DecodeUserId(string hash) => UserHashids.DecodeLong(hash).FirstOrDefault();
}

File: services/auth_rust/src/hashids.rs
use harsh::Harsh;

const PARTY_SALT: &str = "BCRPartyId";
const PARTY_ALPHABET: &str = "ABCDEFGHJKMNPQRSTUVWXYZ0123456789";
const USER_SALT: &str = "BCRUserId";
const USER_ALPHABET: &str = "ABCDEFGHJKMNPQRSTUVWXYZ0123456789";
const MIN_LENGTH: usize = 8;

fn build_harsh(salt: &str, alphabet: &str) -> Harsh {
    Harsh::builder()
        .salt(salt)
        .alphabet(alphabet)
        .length(MIN_LENGTH)
        .build()
        .expect("failed to build Harsh encoder")
}

pub fn encode_party_id(id: i64) -> String {
    let h = build_harsh(PARTY_SALT, PARTY_ALPHABET);
    h.encode(&[id as u64])
}

pub fn encode_user_id(id: i64) -> String {
    let h = build_harsh(USER_SALT, USER_ALPHABET);
    h.encode(&[id as u64])
}
